channeldb/kvdb/interface.go

```diff
@@ -21,12 +21,19 @@ func Update(db Backend, f func(tx RwTx) error) error {
 // View opens a database read transaction and executes the function f with the
 // transaction passed as a parameter. After f exits, the transaction is rolled
 // back. If f errors, its error is returned, not a rollback error (if any
-// occur).
-func View(db Backend, f func(tx RTx) error) error {
+// occur). The passed reset function is called before the start of the
+// transaction and can be used to reset intermediate state. As callers may
+// expect retries of the f closure (depending on the database backend used), the
+// reset function will be called before each retry respectively.
+func View(db Backend, f func(tx RTx) error, reset func()) error {
```

Comment:
May have been a bit laborious to find all the affected sites, but I think this is the simplest way forward in the end.

Reply:
done